source/server/http_dynamic_delay_filter.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include <atomic>
+#include <string>
+
+#include "envoy/server/filter_config.h"
+
+#include "api/server/response_options.pb.h"
+
+namespace Nighthawk {
+namespace Server {
+
+/**
+ * Filter configuration container class for the dynamic delay extension.
+ */
+class HttpDynamicDelayDecoderFilterConfig {
+
+public:
+  HttpDynamicDelayDecoderFilterConfig(nighthawk::server::ResponseOptions proto_config);
+
+  /**
+   * @return const nighthawk::server::ResponseOptions& read-only reference to the proto config
+   * object.
+   */
+  const nighthawk::server::ResponseOptions& server_config() { return server_config_; }
+
+  /**
+   * Increments the number of active instances.
+   */
+  void incrementInstanceCount() { instances()++; }
+  /**
+   * Decrements the number of active instances.
+   */
+  void decrementInstanceCount() { instances()--; }
+  /**
+   * @return uint64_t the approximate number of active instances.
+   */
+  uint64_t approximateInstances() const { return instances(); }
+
+private:
+  const nighthawk::server::ResponseOptions server_config_;
+  static std::atomic<uint64_t>& instances() {
+    // We lazy-init the atomic to avoid static initialization / a fiasco.
+    static std::atomic<uint64_t> a(0);
```

Comment:
MUTABLE_CONSTRUCT_ON_FIRST_USE(std::atomic<uint64>, 0)

What you have is still a fiasco on shutdown.

Reply:
Ugh great catch I’ll amend